Library/Homebrew/cmd/upgrade.rb

```diff
@@ -314,81 +220,88 @@ def depends_on(a, b)
     end
   end
 
-  # @private
-  def formulae_with_runtime_dependencies
-    Formula.installed
-           .map(&:opt_or_installed_prefix_keg)
-           .reject(&:nil?)
-           .reject { |f| f.runtime_dependencies.to_a.empty? }
-  end
-
-  def check_dependents(formulae)
-    return if formulae.empty?
-
-    # First find all the outdated dependents.
-    kegs = formulae_with_runtime_dependencies
-
-    return if kegs.empty?
+  def check_dependents(formulae_to_install)
+    return if formulae_to_install.empty?
 
     oh1 "Checking dependents for outdated formulae" if args.verbose?
-    upgradable, pinned = upgradable_dependents(kegs, formulae).map(&:to_a)
-
-    upgradable.sort! { |a, b| depends_on(a, b) }
-
-    pinned.sort! { |a, b| depends_on(a, b) }
-
-    # Print the pinned dependents.
-    unless pinned.empty?
-      ohai "Not upgrading #{pinned.count} pinned #{"dependent".pluralize(pinned.count)}:"
-      puts pinned.map { |f| "#{f.full_specified_name} #{f.pkg_version}" } * ", "
+    dependents =
+      formulae_to_install.flat_map(&:runtime_installed_formula_dependents)
+    return if dependents.empty?
+
+    upgradeable_dependents = dependents.select(&:outdated?)
```

Comment:
I think this line is causing #6715,  since `dependents` contains kegs, but only a formula has the `outdated?` function defined. The previous code had a `dependent_formulae = dependents.map(&:to_formula)` on line 232. Should this logic be operating on kegs or formulae?

There will be a similar problem on line 240 with pinned_dependents, since keg doesn't have a `full_specified_name` method; only formula has that.